src/dswx_sar/filter_SAR.py

```diff
@@ -122,11 +125,210 @@ def lee_enhanced_filter(img, win_size=3, k=K_DEFAULT, cu=CU_DEFAULT,
     filter_im: numpy.ndarray
         filtered intensity image.
     """
-    print('>> lee_enhanced_filter')
+    print('>> lee_enhanced_filter', kwargs)
+    win_size = kwargs.get('win_size', -1)
 
     # we process the entire img as float64 to avoid type overflow error
     img = np.float64(img)
     w_t, mean, _ = weightingarr(img, win_size, k, cu, cmax)
     filter_im = (mean * w_t) + (img * (1 - w_t))
 
     return filter_im
+
+
+def fill_nan_value(data):
+    mask = np.isnan(data) | np.isinf(data) | np.isinf(-data)
+    data[mask] = -15
+    return data
+
+
+def find_largest_nan_polygon(data):
+    """
+    Identifies contiguous NaN regions and finds the largest one.
+
+    :param data: 2D numpy array with NaNs.
+    :return: The label of the largest NaN region and an array where each
+             contiguous NaN region is labeled with a unique integer.
```

Comment:
```suggestion
    Parameters
    ----------
        data: 2D numpy array with NaNs.
    
    Returns
    -------
    _: tuple:
        The first element being the label of the largest NaN region
        The second element being an array where each contiguous NaN region is labeled with a unique integer.
```

Let's try to use numpy docstring format throughout the code.

Reply:
Similarly, I'll delete this function. It is not needed anymore.